src/pages/workspace/distanceRates/CreateDistanceRatePage.tsx

```diff
@@ -49,7 +49,7 @@ function CreateDistanceRatePage({policy, route}: CreateDistanceRatePageProps) {
     const submit = (values: FormOnyxValues<typeof ONYXKEYS.FORMS.POLICY_CREATE_DISTANCE_RATE_FORM>) => {
         const newRate: Rate = {
             currency,
-            name: CONST.CUSTOM_UNITS.DEFAULT_RATE,
+            name: getOptimisticRateName(customUnits[customUnitID].rates),
```

Comment:
```suggestion
            name: getOptimisticRateName(customUnits[customUnitID]?.rates),
```
Just in case.
Fallback values for customUnits is {} and customUnitID is '' and for these values `customUnits[customUnitID]` will be `undefined`